lib/ansible/plugins/action/__init__.py

```diff
@@ -749,6 +749,16 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
 
         self._update_module_args(module_name, module_args, task_vars)
 
+        # TODO: move this code to a separate function
+        # TODO: parse interpreter from module
+        interpreter = 'python'
+        # TODO: make list of binary names to search for configurable on a per-interpreter basis
+        result = self._low_level_execute_command('which python python3', sudoable=False)
```

Comment:
This will go bang on anything not POSIX (*cough* Windows *cough*)... This behavior, or at least the creation of the command to do it, probably needs to be delegated to the shell plugin- no shell-isms or platform-isms in the ActionBase.

Reply:
Also, this adds a new round-trip for every module exec (at a time when we're already thrashing around trying to improve performance and exec overhead)... There's gotta be a saner way to do this only when necessary; like an upfront check that gets stored, or kickedoff once in response to a module exec failure, and a way to cache it.

Reply:
The intention is to add a per-host per-interpreter cache for these, so that it's not an ongoing performance hit.